blockchain/versionbits.go

```diff
@@ -123,27 +120,45 @@ type deploymentChecker struct {
 // interface.
 var _ thresholdConditionChecker = deploymentChecker{}
 
-// BeginTime returns the unix timestamp for the median block time after which
-// voting on a rule change starts (at the next window).
+// HasEnded returns true if the target consensus rule change has expired
+// or timed out (at the next window).
 //
 // This implementation returns the value defined by the specific deployment the
 // checker is associated with.
 //
 // This is part of the thresholdConditionChecker interface implementation.
-func (c deploymentChecker) BeginTime() uint64 {
-	return c.deployment.StartTime
+func (c deploymentChecker) HasStarted(blkNode *blockNode) bool {
+	deploymentStarter := c.deployment.DeploymentStarter
+	if clockStarter, ok := deploymentStarter.(chaincfg.ClockConsensusDeploymentStarter); ok {
+		clockStarter.SynchronizeClock(c.chain)
+	}
+
+	// Can't fail as we make sure to set the clock above.
+	header := blkNode.Header()
+	started, _ := deploymentStarter.HasStarted(&header)
```

Comment:
Since `deploymentStarter` is an interface, and you don't know how the the actual implementation looks like, wouldn't if be better to check for the error?